File: src/main/kotlin/com/huawei/cangjie/psi/CjBasicType.java
package com.huawei.cangjie.psi;

import com.huawei.cangjie.psi.stubs.CangJieBasicTypeStub;
import com.huawei.cangjie.psi.stubs.elements.CjStubElementTypes;
import com.intellij.lang.ASTNode;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Collections;
import java.util.List;

public class CjBasicType extends CjElementImplStub<CangJieBasicTypeStub> implements CjTypeElement {

//public class CjBasicType extends  CjElementImpl  implements CjTypeElement {


    public CjBasicType(@NotNull ASTNode node) {
        super(node);
    }

    public CjBasicType(@NotNull CangJieBasicTypeStub stub) {
        super(stub, CjStubElementTypes.BASIC_TYPE);
    }

    @Override
    public String toString() {
        return getElementType().toString();
    }

    @Override
    public <R, D> R accept(@NotNull CjVisitor<R, D> visitor, @Nullable D data) {
        return visitor.visitBasicType(this, data);
    }

    @Override
    public String getText() {
        CangJieBasicTypeStub stub = getStub();
        if (stub != null) {
            return stub.getBasicType();


        }
        return super.getText();
    }



    @Override
    @NotNull
    public String getName() {
return getText();
    }


    @NotNull
    public List<CjTypeProjection> getTypeArguments() {
        return Collections.emptyList();

    }

    @Override
    public @NotNull List<CjTypeReference> getTypeArgumentsAsTypes() {

        return Collections.emptyList();
    }
}

File: src/main/kotlin/com/huawei/cangjie/psi/CjBasicType.kt
package com.huawei.cangjie.psi

import com.huawei.cangjie.lexer.CjTokens
import com.huawei.cangjie.name.Name
import com.huawei.cangjie.psi.stubs.CangJieBasicTypeStub
import com.huawei.cangjie.psi.stubs.elements.CjStubElementTypes
import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElement
import com.intellij.psi.tree.IElementType

class CjBasicType : CjElementImplStub<CangJieBasicTypeStub>, CjTypeElement, CjSimpleNameExpression {
    //public class CjBasicType extends  CjElementImpl  implements CjTypeElement {
    constructor(node: ASTNode) : super(node)

    constructor(stub: CangJieBasicTypeStub) : super(stub, CjStubElementTypes.BASIC_TYPE)

    override fun toString(): String {
        return elementType.toString()
    }

    override fun getReferencedName(): String {
        return text
    }

    override fun getReferencedNameAsName(): Name {
        return Name.identifier(this.text)
    }

    override fun getReferencedNameElement(): PsiElement {
        return this
    }

    override fun getIdentifier(): PsiElement {
        return this
    }

    override fun getReferencedNameElementType(): IElementType {
        return this.elementType
    }

    override fun <R, D> accept(visitor: CjVisitor<R, D>, data: D?): R {
        return visitor.visitBasicType(this, data)
    }

    override fun getText(): String {
        val stub = stub
        if (stub != null) {
            return stub.basicType
        }
        return super.getText()
    }


    override fun getName(): String {
        return text
    }


    val typeArguments: List<CjTypeProjection>
        get() = emptyList()

    override val typeArgumentsAsTypes: List<CjTypeReference>
        get() = emptyList()
}
